src/Microsoft.VisualStudio.Threading.Analyzers/VSTHRD200UseAsyncNamingConventionAnalyzer.cs

```diff
@@ -55,7 +55,7 @@ private void AnalyzeNode(SymbolAnalysisContext context)
 
             if (!methodSymbol.Name.EndsWith(MandatoryAsyncSuffix))
             {
-                if (methodSymbol.ReturnType.Name == nameof(Task) &&
+                if ((methodSymbol.ReturnType.Name == nameof(Task) || methodSymbol.ReturnType.Name == "ValueTask") &&
```

Comment:
Wouldn't it be better to support the general case of task-like types?

https://github.com/dotnet/roslyn/blob/master/docs/features/task-types.md

Reply:
This change was meant to be super quick to alleviated an urgent need.
But yes, recognizing the pattern and not hard-coding specific types will avoid this from happening again. *However* the pattern you referenced isn't the one we'll follow, I think, because the key point isn't whether the type returned is compatible with the *async* keyword in a method signature, but rather whether the returned type is awaitable. A great many more types are awaitable than can be used in an *async* method with an async method builder, and so long as it is awaitable, the suffix should be `Async`.